content/_computed/eleventyComputed.js

```diff
@@ -7,7 +7,12 @@ const { warn } = chalkFactory('eleventyComputed')
  * Global computed data
  */
 module.exports = {
-  canonicalURL: ({ publication, page }) => page.url && path.join(publication.url, page.url),
+  canonicalURL: ({ publication, page }) => {
+    const { url } = publication
+    const { origin, pathname } = new URL(url)
+    const pagePath = path.join(pathname, page.url)
+    return new URL(pagePath, origin).href
+  },
```

Comment:
Creating two instances of a URL here may be unneeded, perhaps
```suggestion
  canonicalURL: ({ publication, page }) => {
    const { url } = publication
    return new URL(page.url, url).toString()
  },
```
or even
```
  canonicalURL: ({ publication, page }) => {
    return new URL(page.url, publication.url).toString()
  },
```